### PR TITLE
Shanbady/additional details on offeror channel pages

### PR DESCRIPTION
**Pull request by shanbady**
### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4327

### Description (What does it do?)
This PR adds an info box to offerer pages (/c/offeror/mitpe)

[figma](https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=3774-68962&viewport=182%2C371%2C0.5&t=SgD1ouufEbDt17yz-1&scaling=scale-down&starting-point-node-id=2197%3A31345&show-proto-sidebar=1)

### Screenshots (if appropriate):
Screenshot 2024-06-03 at 1 26 54 PM
Screenshot 2024-06-03 at 1 28 29 PM

### How can this be tested?
1. Checkout this branch
2. make sure you have offerers loaded otherwise run ```python manage.py update_offered_by```
3. go to the different offerer pages http://od.odl.local:8063/c/offeror/xpro http://od.odl.local:8063/c/offeror/mitpe http://od.odl.local:8063/c/offeror/xpro http://od.odl.local:8063/c/offeror/mitx and see that the info box displays the appropriate data for each

### CODE REVIEW

File: frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        : detailValue
+
+      return (
+        <Box key={value.title} sx={{ margin: "10px" }}>
+          <InfoLabel
+            lineHeight="1.5"
+            fontSize="inherit"
+            variant="h5"
+            gutterBottom
+          >
```
Comment (jonkafton) [inline frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx:166]: The designs look like `variant="subtitle2"` for this one (and no `lineHeight` or `fontSize` needed)

File: frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    string[] | string
+  >
+  return data[channelType]
+}
+const InfoLabel = styled(Typography)(({ theme }) => ({
+  color: theme.custom.colors.mitRed,
+}))
+const ChannelDetailsCard = styled(Box)(({ theme }) => ({
+  border: "1px solid #ccc",
+  borderRadius: "8px",
```
Comment (jonkafton) [inline frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx:129]: border-radius: 12px;

File: frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  backgroundColor: "white",
+  minWidth: "300px",
+  width: "50%",
+  fontSize: "0.5em",
+  lineHeight: "0.5em",
+  [theme.breakpoints.down("md")]: {
+    width: "100%",
```
Comment (jonkafton) [inline frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx:136]: The panel isn't occupying its full space on mobile: 

image

File: frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  color: theme.custom.colors.mitRed,
+}))
+const ChannelDetailsCard = styled(Box)(({ theme }) => ({
+  border: "1px solid #ccc",
+  borderRadius: "8px",
+  backgroundColor: "white",
+  minWidth: "300px",
+  width: "50%",
+  fontSize: "0.5em",
+  lineHeight: "0.5em",
```
Comment (jonkafton) [inline frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx:134]: Font sizes handled by nested typography elements - can remove

File: frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    string,
+    string[] | string
+  >
+  return data[channelType]
+}
+const InfoLabel = styled(Typography)(({ theme }) => ({
+  color: theme.custom.colors.mitRed,
+}))
+const ChannelDetailsCard = styled(Box)(({ theme }) => ({
+  border: "1px solid #ccc",
```
[the remark this answers is not in the document]
Comment (shanbady) [reply frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx]: removed

File: frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx (hunk 3 above)
Comment (shanbady) [reply frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx:133]: should look now

File: frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+          >
+            {value.title}:
+          </InfoLabel>
+          <Typography
+            lineHeight="1"
+            fontSize="inherit"
+            variant="body1"
+            color="text.secondary"
+            gutterBottom
+          >
```
[the remark this answers is not in the document]
Comment (shanbady) [reply frontends/mit-open/src/page-components/ChannelDetails/ChannelDetails.tsx]: removed and added variant

**Review by jonkafton — APPROVED** [review]

**Pull request merged**